app/src/main/java/io/intrepid/contest/screens/contestcreation/reviewcontest/ReviewContestFragment.java

```diff
@@ -0,0 +1,77 @@
+package io.intrepid.contest.screens.contestcreation.reviewcontest;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnClick;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseFragment;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.screens.contestcreation.ContestCreationFragment;
+import io.intrepid.contest.screens.contestcreation.EditContestContract;
+import io.intrepid.contest.screens.contestcreation.categorieslist.CategoryAdapter;
+
+public class ReviewContestFragment extends BaseFragment<ReviewContestPresenter> implements ReviewContestContract.View, ContestCreationFragment {
+    private static final String CONTEST_KEY = "CONTEST_KEY";
+    @BindView(R.id.contest_title_button)
+    TextView titleButton;
+    @BindView(R.id.contest_description_button)
+    TextView descriptionButton;
+    @BindView(R.id.categories_recycler_view)
+    RecyclerView categoriesRecyclerView;
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.fragment_review_contest;
+    }
+
+    @NonNull
+    @Override
+    public ReviewContestPresenter createPresenter(PresenterConfiguration configuration) {
+        Contest.Builder contestBuilder = ((EditContestContract) getActivity()).getContestBuilder();
```

Comment:
@kentonguan The architecture refactor I did with @jmccormack200 pretty much leverages the activity and fragment relationship by way of the activity's presenter to track any changes occurring with the contest being built. Wouldn't passing it as a bundle change the behavior of the app?  i.e, the bundled arguments may or may not necessarily match the instance of the contest that the presenter's holding on to. In the previous iterations, I was passing them into each fragment as a bundle and setting up the viewpager again each time a change was made to the contest.